### PR TITLE
Add RGBs for cloud phase (distinction) and type

### LINKED ISSUE #1954: Cloud Phase/Type/Phase Distinction RGBs for VIIRS and FCI
## Feature Request

**Is your feature request related to a problem? Please describe.**

I'd like to generate the 2022 Testbed RGBs with VIIRS:

- Cloud Phase RGB
- Cloud Type RGB
- Cloud Phase Distinction RGB

At least, Cloud Phase and Cloud Phase Distinction already exist for other sensors.  Channel differences should be checked, but I think they can be generated for VIIRS at a 742 meter SSP resolution, as well as with FCI at 2 km (maybe 1 km, need to check) SSP resolution.

**Describe the solution you'd like**

New RGBs.

**Describe any changes to existing user workflow**

N/A

**Additional context**

We can use the VIIRS RGBs to train users preparing them for new FCI RGBs.

### PR DESCRIPTION
**Pull request by gerritholl**
Add three RGBs to the general visir.yaml composite file, sourced to CIRA
and EUMETRAIN.

 - [x] Closes #1954 
 - [ ] Tests added 
 - [x] Fully documented

### CODE REVIEW

**Comment by gerritholl** [comment]
For EARS-VIIRS the `cloud_phase_distinction` RGB tries to use the DNB.  That's not what the doctor ordered.  Should redefine to use `I01` instead, even if that means no more availability for EARS-VIIRS.

**Comment by gerritholl** [comment]
Cloud type
---

Suomi-NPP-cloud_type-nqeuro742m-20211228_1324-1326-geotiff

Cloud phase
---

Suomi-NPP-cloud_phase-nqeuro742m-20211228_1324-1326-geotiff

Cloud phase distinction
---

Suomi-NPP-cloud_phase_distinction-nqeuro742m-20211228_1324-1326-geotiff

**Comment by djhoese** [comment]
How do these compare with the existing ones in the `abi.yaml` file?

https://github.com/pytroll/satpy/blob/2e58eef7ad67f4b20746ba5d22a38a5f5dd0d9da/satpy/etc/composites/abi.yaml#L403-L418

Should the ABI ones be removed? Or should be keep them to guard against the generic `visir.yaml` ones ending up different than the official ABI recipes?

**Comment by mraspaud** [comment]
They look great, thanks for adding these! does the red part in the cloud phase distinction bring something interesting or should it be masked out by default?

**Comment by gerritholl** [comment]
> How do these compare with the existing ones in the `abi.yaml` file?

They're supposed to be identical.  I haven't done a side-by-side comparison.  I would need to find some VIIRS granules covering the ABI disk first.

> Should the ABI ones be removed?

Maybe not.  I haven't tried the generic one with ABI, but I find that Satpy gets confused sometimes.

> Or should be keep them to guard against the generic visir.yaml ones ending up different than the official ABI recipes?

For ABI maybe it doesn't matter.  For VIIRS I ended up adding a specific one after adding the one to `visir.yaml` because Satpy was trying to use the DNB, which was not intended.

> Does the red part in the cloud phase distinction bring something interesting or should it be masked out by default?

I don't think it brings something interesting.  The cloud phase distinction is a daytime only composite.  Should we mask it out?  I'm not sure; I took what exists in Satpy for AHI and ABI, which don't mask it out either.

**Review by djhoese — COMMENTED** [review]
Small question on rayleigh correction, but otherwise this looks good to me.

File: satpy/etc/composites/viirs.yaml [hunk 1]
```diff
@@ -547,3 +547,81 @@ composites:
...
+      CIRA Quick Guide: https://rammb.cira.colostate.edu/training/visit/quick_guides/QuickGuide_DayCloudPhaseDistinction_final_v2.pdf
+      Cloud Type recipe and typical colours: https://www.eumetsat.int/fr/media/45659
+    ## it uses the default used in etc/enhancements/generic.yaml
+    compositor: !!python/name:satpy.composites.GenericCompositor
+    prerequisites:
+      - name: M15
+      - name: I01
+        modifiers: [sunz_corrected_iband, rayleigh_corrected]
+      - name: I03
+        modifiers: [sunz_corrected_iband]
```
[the remark this answers is not in the document]
Comment (djhoese) [reply satpy/etc/composites/viirs.yaml:565]: Ah I didn't realize I03 was that wavelength. Yeah that may be getting in the range where pyspectral doesn't apply any adjustment.

**Comment by gerritholl** [comment]
I realised that I'm not sure if I've implemented cloud type and cloud phase correctly.  It's not easy to compare directly with ABI because I don't have VIIRS data in the ABI region.

**Comment by gerritholl** [comment]
The cloud type and cloud phase RGBs were indeed missing their enhancements.

### Cloud Type RGB, without enhancement
202201181408-NOAA-20-viirs-eurol-cloud_type-unenhanced

### Cloud type RGB, with enhancement
202201181408-NOAA-20-viirs-eurol-cloud_type-enhanced

### Cloud Phase RGB, without enhancement
202201181408-NOAA-20-viirs-eurol-cloud_phase-unenhanced

### Cloud Phase RGB, with enhancement
202201181408-NOAA-20-viirs-eurol-cloud_phase-enhanced

The latest commits add the enhancements.

File: satpy/etc/composites/viirs.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/etc/composites/viirs.yaml:565]: I03 is 1.6 micron, so indeed rayleigh doesn't really apply there

File: satpy/etc/composites/viirs.yaml [hunk 2]
```diff
@@ -547,3 +547,81 @@ composites:
...
+    description: >
+      same as cloud_phase RGB product, without modifiers
+    compositor: !!python/name:satpy.composites.GenericCompositor
+    prerequisites:
+      - name: I03
+      - name: M11
+      - name: M05
+    standard_name: cloud_phase
+
+  cloud_type:
```
[the remark this answers is not in the document]
Comment (gerritholl) [reply satpy/etc/composites/viirs.yaml:604]: I could call it `cimss_cloud_type`?

[the remark this answers is not in the document]
Comment (gerritholl) [reply satpy/etc/composites/viirs.yaml]: Renamed `cimss_cloud_type`.

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**